### PR TITLE
Fix addon blockmenu loading with custom worlds

### LINKED ISSUE #2862: Inventory issues when using addons and certain world loading orders
## :round_pushpin: Description (REQUIRED)
Im sorry about my really bad english in never learned it at school so its hard for me to explain something :'( if you have some tips for my english im open about suggestions :D
Hello, I have an issue with slimefun (slimefun addons for more info) i talked with fluffy and riley about this but they saying its something from storage on slimefun itself.

When i put a Void storage from IX , if the server restart i cannot open the barel anymore, i have to break and replace it to access to the inventory (there is a screenshot of the /sf debug_fish when the inventory doesn't work)

if i remove bentobox and then restart the server it works again but if i add it again it wont.

there is a video where we can see the machine working after breaking and replace and no error in console :
https://www.youtube.com/watch?v=17kT6xaPRcQ&feature=youtu.be

I tested another things , i removed placeholder API from plugin list and then restarted and the machines was working again  there is the video : https://www.youtube.com/watch?v=K0ucZLZSDUI&feature=youtu.be

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

1. add all of plugin on the screen under this section
2. put machine from liteExpansion or FluffyMachine 
3. restart the server 
4.try to open machine it will not open and with debut fish we can see "X" on inventory
image
Dont worry about custom names i also tried with officials versions

## :bulb: Expected behavior (REQUIRED)
the correct behavior is when i restart the server all machine still works (no need to break and replace it after every daily restarts)

## :scroll: Server Log
https://pastebin.com/3aCb0KJW
No error on console when openning or breaking

## :open_file_folder: /error-reports/ Folder
The file is empty

## :compass: Environment (REQUIRED)
Tuinity git-Tuinity-"7734ef0" (MC: 1.16.5)
Slimefun DEV - 825 (git 49950cf6)
Metrics-Module #21
Java 15

Installed Addons: (6)
  DyedBackpacks vDEV - 13 (git 964b2d55)
  InfinityExpansion vDEV - 86 (git 034449ee)
  ExtraTools vDEV - 24 (git 791c7468)
  LiteXpansion vDEV - 131 (git bed585a7)
  FluffyMachines vDEV - 67 (git d5ed4a58)
  ExoticGarden vDEV - 73 (git ae423e3e)

 - Server Software: Tuinity git-Tuinity-"7734ef0" (MC: 1.16.5)
 - Minecraft Version: 1.16.5
 - Slimefun Version: dev 825

### PR DESCRIPTION
**Pull request by Mooy1**
## Description
When custom worlds were loaded before a certain addon had been loaded, the BlockMenuPresets for the stored inventories couldn't be found, so the block had no inventory. When the listener is registered after all addons load, it allows all the presets to be loaded before inventories are. The BlockStorages for the worlds are still created because Slimefun is already iterating through all of the default ones during the startup task.

## Changes
Moved WorldListener creation from registerListeners to SlimefunStartupTask

## Related Issues
Resolves #2862 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/SlimefunStartupTask.java [hunk 1]
```diff
@@ -56,6 +57,9 @@ public void run() {
                 SlimefunPlugin.logger().log(Level.SEVERE, x, () -> "An Error occurred while trying to load World \"" + world.getName() + "\" for Slimefun v" + SlimefunPlugin.getVersion());
             }
         }
+        
+        // Load/Unload Worlds, only after all plugins have started up.
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/SlimefunStartupTask.java:61]: Add a "Fixes #2862" to this so we know in the future why it's here

```suggestion
        // Load/Unload Worlds, only after all plugins have started up. Fixes #2862 
```

**Review by WalshyDev — APPROVED** [review]
Not sure this is the best way to resolve this issue but it works until the storage re-write is done

**Review by WalshyDev — APPROVED** [review]
LGTM

**Comment by amarcais53** [comment]
i tested these changes on my server and for the moment everything good

**Review by LinoxGH — APPROVED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**